include/soci/once-temp-type.h

```diff
@@ -31,7 +31,7 @@ class SOCI_DECL once_temp_type
     once_temp_type(once_temp_type const & o);
     once_temp_type & operator=(once_temp_type const & o);
 
-    ~once_temp_type() SOCI_NOEXCEPT_FALSE;
+    ~once_temp_type() noexcept(false);
```

Comment:
Afaik `noexcept(false)` hass absolutely no effect. The boolean argument to `noexcept` only has meaning in a metaprogramming context where a function is conditionally `noexcept`. But if your ahrd-code `false` here, it seems it would be better to have no noexcpet statement at all :thinking:

Reply:
Sorry, this is simply not true. Destructors are noexcept by default in C++11 and we absolutely need this here as this dtor can throw.

Reply:
Ah whoops - I missed that this is a dtor.